example/src/main/java/org/xmtp/android/example/connect/ConnectWalletViewModel.kt

```diff
@@ -31,10 +46,57 @@ class ConnectWalletViewModel : ViewModel() {
         }
     }
 
+    @UiThread
+    fun connectWallet() {
+        viewModelScope.launch(Dispatchers.IO) {
+            _uiState.value = ConnectUiState.Loading
+            try {
+                val peerMeta = WCPeerMeta(name = "XMTP Example", url = "https://xmtp.org")
+                val key = ByteArray(32).also { Random().nextBytes(it) }.toNoPrefixHexString()
+                val bridge = URLEncoder.encode("https://safe-walletconnect.safe.global/", "UTF-8")
+                val wcURI = "wc:${UUID.randomUUID()}@1?bridge=$bridge&key=$key"
+
+                val session = WCSession.from(wcURI) ?: throw InvalidSessionException()
+                val wcClient = WCClient(gsonBuilder, okHttpClient)
+                wcClient.onDisconnect = { _, _ ->
+                    Log.e("###", "DISCONNECT")
+                    if (wcClient.session != null) {
+                        wcClient.killSession()
+                    } else {
+                        wcClient.disconnect()
+                    }
+                }
+                wcClient.onSignTransaction = { _, _->
+                    Log.e("###", "SIGN")
+                }
+                wcClient.onSessionRequest = { _, peer ->
+                    Log.e("###", "SESSION REQ PEER: $peer")
+                }
+                wcClient.onFailure = { t ->
+                    Log.e("###", "FAILURE")
+                }
+                wcClient.connect(session, peerMeta)
+                _uiState.value = ConnectUiState.Connect(wcURI)
+                for (i in 0 .. 30) {
+                    if (wcClient.isConnected) {
+                        Log.e("###", "CONNECTED!")
```

Comment:
I think I might have to try a different WC library since we can't seem to query for address or implement signatures here in order to create a `SigningKey`

Reply:
Love it! Will give it a shot.